starlite/provide.py

```diff
@@ -11,17 +11,19 @@
 
 
 class Provide:
-    __slots__ = ("dependency", "use_cache", "value", "signature_model", "sync_to_thread")
+    __slots__ = ("dependency", "use_cache", "value", "signature_model", "sync_to_thread", "is_coro")
 
     def __init__(self, dependency: AnyCallable, use_cache: bool = False, sync_to_thread: bool = False):
         self.dependency = dependency
         self.use_cache = use_cache
         self.value: Any = Undefined
         self.signature_model: Optional[Type[SignatureModel]] = None
         self.sync_to_thread = sync_to_thread
-        if ismethod(dependency) and hasattr(dependency, "__self__"):
-            # ensure that the method's self argument is preserved
-            self.dependency = partial(dependency, dependency.__self__)
+        # manage async partial objects in 3.7 (https://stackoverflow.com/a/52422903/6560549)
+        if isinstance(dependency, partial):
+            self.is_coro = iscoroutinefunction(dependency.func)
```

Comment:
I prefer being explicit rather than using aliases, so `is_coroutine` instead of `is_coro`